### PR TITLE
ref(crons): Set null for owner_user_id when a user is deleted

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/monitors/models.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ class Monitor(Model):
     Type of monitor. Currently there are only CRON_JOB monitors.
     """
 
-    owner_user_id = HybridCloudForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete="CASCADE")
+    owner_user_id = HybridCloudForeignKey(settings.AUTH_USER_MODEL, null=True, on_delete="SET_NULL")
```
Comment (wedamija) [inline src/sentry/monitors/models.py:254]: Nit: I think maybe it's better to use `HybridCloudForeignKeyCascadeBehavior.SET_NULL`. Both are valid, but it's easier to see all the options for other people looking for examples

**Review by wedamija — APPROVED** [review]
Not sure but I assume you need to generate a migration here

File: src/sentry/monitors/models.py (hunk 1 above)
Comment (evanpurkhiser) [reply src/sentry/monitors/models.py:254]: Yeah but we don't do it anywhere else lol

Comment (wedamija) [reply src/sentry/monitors/models.py:254]: 🥲

Comment (evanpurkhiser) [reply src/sentry/monitors/models.py:254]: let me clean up

**Review by wedamija — APPROVED** [review]

**Review by markstory — APPROVED** [review]
Looks good to me. When a user is deleted, any owner_user_id with a matching value will be nulled out eventually by tombstones.

**Pull request merged**